File: frontend/front-srv/assets/gui.ajax/res/js/ui/ReactUI/router/Router.js
/*
 * Copyright 2007-2017 Charles du Jeu - Abstrium SAS <team (at) pyd.io>
 * This file is part of Pydio.
 *
 * Pydio is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Pydio is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with Pydio.  If not, see <http://www.gnu.org/licenses/>.
 *
 * The latest code can be found at <https://pydio.com>.
 */

import {Switch} from 'react-router-dom';
import Router from 'react-router/lib/Router';
import Route from 'react-router/lib/Route';
import IndexRoute from 'react-router/lib/IndexRoute';
import browserHistory from 'react-router/lib/browserHistory';

import MainRouter from './MainRouter';
import WorkspaceRouter from './WorkspaceRouter';
import PathRouter from './PathRouter';
import {OAuthLoginRouter, OAuthConsentRouter} from './OAuthRouter';
import LoginRouter from './LoginRouter';
import LoginCallbackRouter from './LoginCallbackRouter';
import LogoutRouter from './LogoutRouter';
import TestRouter from './TestRouter';

function getRoutes(pydio){
    const routes = (
        <Switch>
            <Route path="login">
                <IndexRoute component={LoginRouter(pydio)}/>
                <Route path="callback" component={LoginCallbackRouter(pydio)} />
            </Route>
            <Route path="test" component={TestRouter(pydio)} />
            <Route path="logout" component={LogoutRouter(pydio)} />
            <Route path="oauth2">
                <Route path="login" component={OAuthLoginRouter(pydio)} />
                <Route path="consent" component={OAuthConsentRouter(pydio)} />
            </Route> 
            <Route path="/" component={MainRouter(pydio)}>
                <Route path=":workspaceId" component={WorkspaceRouter(pydio)}>
                    <IndexRoute component={PathRouter(pydio)}/>
                    <Route path="*" component={PathRouter(pydio)}/>
                </Route>
            </Route>
        </Switch>
    )
    return routes;
}

class PydioRouter extends React.PureComponent {
    constructor(props) {
        super(props)

        this.state = {
            renderOnce: true
        }
    }

    render() {
        return (
            // Routes are defined as a constant to avoid warning about hot reloading
            <Router history={browserHistory} routes={getRoutes(this.props.pydio)} />
        )
    }
}

export default PydioRouter;

File: idm/authory/web/router.go
/*
 * Copyright (c) 2018. Abstrium SAS <team (at) pydio.com>
 * This file is part of Pydio Cells.
 *
 * Pydio Cells is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Pydio Cells is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with Pydio Cells.  If not, see <http://www.gnu.org/licenses/>.
 *
 * The latest code can be found at <https://pydio.com>.
 */

package web

import (
	"net/http"

	"github.com/gorilla/mux"
)

type route struct {
	name        string
	method      string
	pattern     string
	handlerFunc http.HandlerFunc
}

type routes []route

// NewRouter creates and configures a new mux router to serve wopi REST requests and enable integration with WOPI clients.
func NewRouter() *mux.Router {
	router := mux.NewRouter().StrictSlash(true)
	for _, route := range myRoutes {
		var handler http.Handler
		handler = route.handlerFunc
		handler = logger(handler, route.name)
		// handler = auth(handler)

		router.
			Methods(route.method).
			PathPrefix("/auth/dex").
			Path(route.pattern).
			Name(route.name).
			Handler(handler)
	}

	return router
}

var myRoutes = routes{
	route{
		"Login",
		"POST",
		"/login",
		login,
	},

	route{
		"Login callback",
		"GET",
		"/login",
		loginCallback,
	},

	route{
		"Consent callback",
		"GET",
		"/consent",
		consentCallback,
	},

	route{
		"Login Connector",
		"GET",
		"/login/{connector}",
		loginConnector,
	},

	route{
		"Login Connector callback",
		"GET",
		"/login/{connector}/callback",
		loginConnectorCallback,
	},

	route{
		"Login Connector callback",
		"POST",
		"/login/{connector}/callback",
		loginConnectorCallback,
	},

	route{
		"Consent",
		"GET",
		"/callback",
		callback,
	},

	route{
		"Logout",
		"GET",
		"/logout",
		logout,
	},
}
